langchain4j-github-models/src/main/java/dev/langchain4j/model/github/InternalGitHubModelHelper.java

```diff
@@ -0,0 +1,365 @@
+        return chatCompletionsClientBuilder.buildAsyncClient();
+    }
+
+    private static ChatCompletionsClientBuilder setupChatCompletionsClientBuilder(String endpoint, String serviceVersion, String gitHubToken, Duration timeout, Integer maxRetries, ProxyOptions proxyOptions, boolean logRequestsAndResponses, String userAgentSuffix, Map<String, String> customHeaders) {
+
+        String endpointToUse = isNullOrBlank(endpoint) ? DEFAULT_GITHUB_MODELS_ENDPOINT : endpoint;
+
+        timeout = getOrDefault(timeout, ofSeconds(60));
+        HttpClientOptions clientOptions = new HttpClientOptions();
+        clientOptions.setConnectTimeout(timeout);
+        clientOptions.setResponseTimeout(timeout);
+        clientOptions.setReadTimeout(timeout);
+        clientOptions.setWriteTimeout(timeout);
+        clientOptions.setProxyOptions(proxyOptions);
+
+        String userAgent = DEFAULT_USER_AGENT;
+        if (userAgentSuffix!=null && !userAgentSuffix.isEmpty()) {
+            userAgent = DEFAULT_USER_AGENT + "-" + userAgentSuffix;
+        }
+        List<Header> headers = new ArrayList<>();
+        headers.add(new Header("User-Agent", userAgent));
+        if (customHeaders != null) {
+            customHeaders.forEach((name, value) -> headers.add(new Header(name, value)));
+        }
+        clientOptions.setHeaders(headers);
+        HttpClient httpClient = new NettyAsyncHttpClientProvider().createInstance(clientOptions);
+
+        HttpLogOptions httpLogOptions = new HttpLogOptions();
+        if (logRequestsAndResponses) {
+            httpLogOptions.setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS);
+        }
+
+        maxRetries = getOrDefault(maxRetries, 3);
+        ExponentialBackoffOptions exponentialBackoffOptions = new ExponentialBackoffOptions();
+        exponentialBackoffOptions.setMaxRetries(maxRetries);
+        RetryOptions retryOptions = new RetryOptions(exponentialBackoffOptions);
+
+        ChatCompletionsClientBuilder chatCompletionsClientBuilder = new ChatCompletionsClientBuilder()
+                .endpoint(endpointToUse)
+                .serviceVersion(getModelServiceVersion(serviceVersion))
+                .httpClient(httpClient)
+                .clientOptions(clientOptions)
+                .httpLogOptions(httpLogOptions)
+                .retryOptions(retryOptions);
+
+        if (gitHubToken != null) {
+            chatCompletionsClientBuilder.credential(new AzureKeyCredential(gitHubToken));
+        } else {
+            throw new IllegalArgumentException("GitHub token is a mandatory parameter for connecting to GitHub models.");
+        }
+
+        return chatCompletionsClientBuilder;
+
+    }
+
+    public static ModelServiceVersion getModelServiceVersion(String serviceVersion) {
```

Comment:
The idea (and this is copied from Azure OpenAI), is that if the user writes a version higher than the actual one, it will fallback to the latest model available. Then, when there's a library upgrade, they will get the new version without refactoring.
I'm not expecting people to use this a lot, as users have no control over the versions deployed on GitHub Models.

Reply:
OK, I have changed it to use ModelServiceVersion directly